### DISCUSSION TITLE
WARNING:root:Invalid model URL, using default initialization

### DISCUSSION THREAD

**Discussion by akarshsoman** [mindee/doctr#890] [General]
when i try to run this code , it shows WARNING:root:Invalid model URL, using default initialization, can u please help me with this
from doctr.models import ocr_predictor
from doctr.io import DocumentFile

model = ocr_predictor(reco_arch='sar_resnet31', pretrained=True)
multi_img_doc = DocumentFile.from_images(["2.jpg", "4.jpg"])
result = model(multi_img_doc)
result.show(multi_img_doc)
result

**Comment by felixdittrich92** [comment]
@akarshsoman Hi 👋 ,
which framework do you use ? (PyTorch has currently no pretrained for sar_resnet31)

**Reply by akarshsoman** [reply]
i was using pytorch. Thanks

**Reply by khawar-islam** [reply]
Same situation, the processing speed is very slow

**Reply by felixdittrich92** [reply]
Hi @khawar-islam :wave:,
SAR implementation is fixed on main in both frameworks but has still no pretrained version we will add this in v0.6 :)
But if you are looking for fast latency on CPU you should use models which use mobilenet as backbone these are much lighter. 
SAR and MASTER works really well but yeah i would suggest to use both models only on GPU

**Reply by frgfm** [reply]
To add on Felix's answer, this discussion is not related to latency @khawar-islam  :thinking:
Let's keep things in their corresponding threads/discussions/issues, this discussion is about a warning for parameter initialization when instantiating some models!

**Reply by aditya-ihx** [reply]
I still get this error. 
I am on v0.7.0.


```python
model = ocr_predictor(det_arch='db_resnet50', reco_arch='parseq', pretrained=True)
# # PDF
doc = DocumentFile.from_images("103291164/page_1.jpeg")
# # Analyze
result = model(doc)
```
Output is `WARNING:root:Invalid model URL, using default initialization.`

If this is a problem with pretrained version, can I please know which models do have functional pretrained versions. Could not find such a thing in the docs

**Reply by felixdittrich92** [reply]
Hi @aditya-ihx 

v0.7.0:

TF: all models have pretrained checkpoints

PT: 
Detection:
db_resnet50, db_mobilenet_v3_large, db_resnet50_rotation
Recognition:
crnn_vgg16_bn, crnn_mobilenet_v3_small, crnn_mobilenet_v3_large


main branch: (v0.7.1a0 currently)

PT:
Recognition:
master, sar_resnet31, vitstr_small, vitstr_base, parseq


We are still busy with training runs but with the next release each model will have a pretrained checkpoint :)

**Reply by aditya-ihx** [reply]
Since parseq for PT is available, what would be the reason I get the warning `WARNING:root:Invalid model URL, using default initialization.`

Thanks in advance

**Reply by felixdittrich92** [reply]
You mentioned that you are on v0.7.0 atm. but parseq is only available at the main branch :) (v0.7.1)
So for parseq you need to clone the repo and install it locally

**Reply by aditya-ihx** [reply]
Oh! I am sorry. That was a pretty dumb question.

Thank you so much :)

**Comment by mehdi-jamaseb** [comment]
I hope this message finds you well. I am currently working on an OCR project using the doctr library and encountered an issue when trying to use the fast_tiny detection architecture. Despite setting pretrained=True, I receive the following warning:

WARNING:root:Invalid model URL, using default initialization.
Here is the relevant part of my code:

from doctr.models import ocr_predictor

det_arch = "fast_tiny"
rec_arch = "crnn_vgg16_bn"
predictor = ocr_predictor(det_arch=det_arch, reco_arch=rec_arch, pretrained=True)

Could you please advise on how I can resolve this issue so that the pretrained model is correctly downloaded and used? I have ensured that I am using the latest version of doctr.

Thank you in advance for your assistance

**Reply by SkaarFacee** [reply]
Can you also state what version of docTR you are using ?

**Reply by felixdittrich92** [reply]
Hi @mehdi-jamaseb 👋,

As mentioned in the last release notes the `fast` checkpoints will be official available with the next release (v0.9.0).
So if you want to test it yet you need to switch to the main branch (v0.9.0a0)

Best regards,
Felix

**Reply by SkaarFacee** [reply]
@felixdittrich92 Yep, thanks for that. I wasn't sure what version had the `fast` checkpoints. That is why I asked about the version. Definitely seemed like a version issue.

**Reply by mehdi-jamaseb** [reply]
Dear @felixdittrich92 ,
Thank you for your response and the information regarding the availability of the fast checkpoints in the upcoming v0.9.0 release.
I am currently using Colab for my project. Could you please provide guidance on how I can switch to the main branch (v0.9.0a0) in Colab to test the fast checkpoints?
Additionally, if I were to manually update the URL for the pretrained model, could you please advise on the steps or the specific URL format I should use?
Thank you once again for your assistance.

**Reply by mehdi-jamaseb** [reply]
thanks @SkaarFacee

**Reply by felixT2K** [reply]
@mehdi-jamaseb 
In colab you can install from the main branch with

```
!pip3 install python-doctr[torch]@git+https://github.com/mindee/doctr.git
```

to load custom models please take a look at:
https://mindee.github.io/doctr/using_doctr/custom_models_training.html#loading-your-custom-trained-model

So you would need first to download the model and then load as described in the docs :)

PS: the fast model has a minor bug in v0.8.1 so i suggest to use the main branch ftm

**Reply by mehdi-jamaseb** [reply]
@felixT2K 
Thank you for the instructions.
Thanks again for your help and the bug notice in v0.8.1

**Reply by felixT2K** [reply]
If you are looking for something more lightweight you could also take a look at: https://github.com/felixdittrich92/OnnxTR 
(the last release contains already all features from the current doctr main branch)
